modules/azure/common_test.go

```diff
@@ -0,0 +1,61 @@
+package azure
+
+import "testing"
+
+func Test_getTargetAzureSubscription(t *testing.T) {
+	type args struct {
+		subID string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{name: "subIDProvidedAsArg", args: args{subID: "test"}, want: "test", wantErr: false},
+		{name: "subIDNotProvided", args: args{subID: ""}, want: "", wantErr: true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := getTargetAzureSubscription(tt.args.subID)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("getTargetAzureSubscription() error = %v, wantErr %v", err, tt.wantErr)
```

Comment:
Seems safer to check for a specific error. See `assert.Error` and `assert.NoError`.